lib/semantics/check-do.cc

```diff
@@ -28,12 +28,34 @@ namespace Fortran::semantics {
 
 using namespace parser::literals;
 
+static NamePtr GetNamePtr(const std::optional<parser::Name> &name) {
+  if (name.has_value()) {
+    return &(name.value());
```

Comment:
You can use `*` rather than `.value()` immediately after testing for the presence of the value if you like.

You don't have to parenthesize the operand of a prefix operator in C/C++ if the operand is a primary or an application of a postfix operator.  All postfix operators in C/C++ bind more tightly than any prefix operator does.

Reply:
Thanks, Peter.  I purposely chose to use the ```.value()``` because I thought that using ```*``` might look like an ordinary pointer dereference, especially since I was in the process of creating a pointer.  I'll remove the parentheses.